Java/core/src/main/java/com/amazon/randomcutforest/sampler/AbstractStreamSampler.java

```diff
@@ -158,4 +184,91 @@ public void setMostRecentTimeDecayUpdate(long index) {
             throw new IllegalArgumentException("Unsupported configuration setting: " + name);
         }
     }
+
+    /**
+     * @return the number of points contained by the sampler when full.
+     */
+    @Override
+    public int getCapacity() {
+        return capacity;
+    }
+
+    public double getInitialAcceptFraction() {
+        return initialAcceptFraction;
+    }
+
+    public long getRandomSeed() {
+        return random.randomSeed;
+    }
+
+    protected class ReplayableRandom {
+        long randomSeed;
+        Random testRandom;
+
+        ReplayableRandom(long randomSeed) {
+            this.randomSeed = randomSeed;
+        }
+
+        ReplayableRandom(Random random) {
+            this.testRandom = random;
+        }
+
+        double nextDouble() {
+            if (testRandom != null) {
+                return testRandom.nextDouble();
+            }
+            Random newRandom = new Random(randomSeed);
+            randomSeed = newRandom.nextLong();
+            return newRandom.nextDouble();
+        }
+    }
+
+    public static class Builder<T extends Builder<T>> {
+
+        // We use Optional types for optional primitive fields when it doesn't make
+        // sense to use a constant default.
+
+        protected int capacity = DEFAULT_SAMPLE_SIZE;
+        protected double timeDecay = 0;
+        protected Random random = null;
+        protected long randomSeed;
+        protected long maxSequenceIndex = 0;
+        protected long sequenceIndexOfMostRecentLambdaUpdate = 0;
+        protected double initialAcceptFraction = DEFAULT_INITIAL_ACCEPT_FRACTION;
+
+        public T capacity(int capacity) {
+            this.capacity = capacity;
+            return (T) this;
+        }
+
+        public T randomSeed(long seed) {
+            this.randomSeed = seed;
+            return (T) this;
+        }
+
+        public T random(Random random) {
+            this.random = random;
+            return (T) this;
+        }
+
+        public T maxSequenceIndex(long maxSequenceIndex) {
+            this.maxSequenceIndex = maxSequenceIndex;
+            return (T) this;
+        }
+
+        public T mostRecentLambdaUpdate(long sequenceIndexOfMostRecentLambdaUpdate) {
```

Comment:
`Lambda` should be `TimeDecay` to match.

Reply:
Done. We probably should have a more focused PR to have all lambda -> timeDecay elsewhere.